geniza/corpus/management/commands/export_annotations.py

```diff
@@ -0,0 +1,143 @@
+        transcription_output_dir = {}
+        for output_format in ["txt", "html"]:
+            print([settings.ANNOTATION_BACKUP_PATH, "transcriptions", output_format])
+            format_path = os.path.join(
+                settings.ANNOTATION_BACKUP_PATH, "transcriptions", output_format
+            )
+            transcription_output_dir[output_format] = format_path
+            os.makedirs(format_path, exist_ok=True)
+
+        # identify content to backup based on documents with digital editions
+        docs = Document.objects.filter(
+            footnotes__doc_relation__contains=Footnote.DIGITAL_EDITION
+        ).distinct()
+        self.stdout.write(
+            "Backing up annotations for %d documents with digital editions"
+            % docs.count()
+        )
+        for document in docs[:3]:
+            print(document)
+            # use PGPID for annotation directory name
+            # path based on recommended uri pattern from the spec
+            # {prefix}/{identifier}/list/{name}
+            output_dir = os.path.join(annotations_output_dir, str(document.id), "list")
+            # ensure output directory exists
+            os.makedirs(output_dir, exist_ok=True)
+
+            # find all annotations for this document
+            annos = Annotation.objects.filter(
+                content__target__source__partOf__id=document.manifest_uri
+            )
+            # TODO: ensure ordered by position / date created
+            # (updated from latest view logic)
+
+            # aggregate annotations by canvas id
+            # for annotation list backup, we don't need to segment
+            # by motivation, source, etc; just group by canvas
+            annos_by_canvas = defaultdict(list)
+            for a in annos:
+                annos_by_canvas[a.target_source_id].append(a)
+
+            # create and save one AnnotationList per canvas
+            # with all associated annotations, as a backup
+            # that could be imported into any w3c annotation server
+
+            for canvas, annotations in annos_by_canvas.items():
+                annolist_name = self.annotation_list_name(canvas)
+                annolist_out_path = os.path.join(output_dir, "%s.json" % annolist_name)
+                print(annolist_out_path)
+                with open(annolist_out_path, "w") as outfile:
+                    json.dump(
+                        annotations_to_list(annotations, uri="test"), outfile, indent=2
+                    )
+
+            # for convenience and more readable versioning, also generate
+            # text and html transcription files
+            for edition in document.digital_editions():
+                print(edition)
+                # filename should be based on pgpid and source
+                # use source id for now; slug todo
+                # should we explicitly label as transcription?
+                base_filename = "%s_source_%s" % (document.id, edition.source.id)
```

Comment:
I've been mulling it over — implementing slugs for sources seems like way too much overhead for right now, and still potentially brittle.  I think using slugified author last names should be sufficient, since we shouldn't have more than one transcription for a document by the same unique set of authors.  

Proposed filename format:
PGPID###_authorslug_transcription.txt (or .html)

Reply:
That proposal looks good to me, agree it's better to slugify by authors.

Reply:
> Ah, OK, re-reading this I understand now. Living in the `Document` model and firing on a signal when the annotation changes makes sense to me, since we want it grouped first by document and then by canvas.

I was thinking we will want to run the backup two ways:
- script that updates everything or specific documents by id; could be run as a cron job on all documents to make sure everything is up to date (should do nothing if there are no changes)
- signal handler that updates annotation exports for a single document when annotations change (I'm a little worried about the signal handler delays already, which was why I was starting to look into async task queues, but I'd prefer to keep things simpler if possible). 

We'll need logic to delete outdated transcription files; maybe that only happens when the full script runs?